roadmap/implementers-guide/src/runtime/disputes.md

```diff
@@ -17,90 +19,351 @@ We account for these requirements by having the validity module handle two kinds
 1. Local disputes: those contesting the validity of the current fork by disputing a parablock included within it.
 1. Remote disputes: a dispute that has partially or fully resolved on another fork which is transplanted to the local fork for completion and eventual slashing.
 
-## Approval
+## Information Consumption from Availability
 
-We begin approval checks upon any candidate immediately once it becomes available.  
+There is little overlap between the approval system and the disputes systems since disputes cares only that two validators disagree.
+However it is required that disputes count validity votes from elsewhere, both the backing votes and the approval votes.
+This is necessary to slash the losing party appropriately which is based on the voting behaviour and resolution.
 
-Assigning approval checks involve VRF secret keys held by every validator, making it primarily an off-chain process.  All assignment criteria require specific data called "stories" about the relay chain block in which the candidate assigned by that criteria became available.  Among these criteria, the BABE VRF output provides the story for two, and the other's story consists of the candidate's block hash plus external knowledge that a relay chain equivocation exists with a conflicting candidate. 
+## Local Disputes
 
-We liberate availability cores when their candidate becomes available of course, but one approval assignment criteria continues associating each candidate with the core number it occupied when it became available. 
+Who does local disputes, validators self-select based on a [VRF](##-Secondary-Approval-Checking).
 
-Assignment proceeds in loosely timed rounds called `DelayTranche`s roughly 12 times faster than block production, in which validators send assignment notices until all candidates have enough checkers assigned.  Assignment tracks when approval votes arrive too and assigns more checkers if some checkers run late.
+We could approve, and even finalize,
+a relay chain block which then later disputes
+due to claims of some parachain being invalid.
```

Comment:
Yes, I suppose this might be a "local only dispute" in the sense that we're reverting finality and that'll be contentious.  We've various mitigations that folks could explore later, but initially maybe switch the relay chain into some "maintenance mode" where governance sorts it out, like if a bad upgrade causes this then that sounds better.  Anywho..

Reply:
I think we need to clarify the consequences of that in more detail, and if we even have such a mode and what that would entail including side effects.

Reply:
I'm afraid I've brain dumped this paper after every time I spoke about it with folks Al, but..

Initially we'll encounter this situation from mostly bugs, so a maintenance mode that does not yet revert sound simplest, and if reversion is required then maybe governance could manually post the same revert transaction that gets posted automatically if not finalized.  

A detail is if "maintenance mode" shuts down all parachains or only the effected parachains plus any "dangerous" parachains like bridges, probably all initially.  I think "maintenance mode" would permit other relay chain activity besides parachains, like code upgrade, transactions, staking, and even continue advancing finality, but anyone who watches finality needs some notification that finality might be reverted and even possibly to where it might be reverted.